cmd/operator/main.go

```diff
@@ -202,6 +209,28 @@ func run(fs *flag.FlagSet) int {
 		return 1
 	}
 
+	if ok := operator.HasFeatureGate("auto-gomemlimit"); ok {
+		if memlimitRatio >= 1.0 {
+			memlimitRatio = 1.0
+		} else if memlimitRatio <= 0.0 {
+			memlimitRatio = 0.0
+		}
+
+		if _, err := memlimit.SetGoMemLimitWithOpts(
+			memlimit.WithRatio(memlimitRatio),
+			memlimit.WithProvider(
+				memlimit.ApplyFallback(
+					memlimit.FromCgroup,
+					memlimit.FromSystem,
+				),
+			),
+		); err != nil {
+			level.Warn(logger).Log("component", "automemlimit", "msg", "Failed to set GOMEMLIMIT automatically", "err", err)
+		} else {
```

Comment:
Are we adding automatic gomemlimit to other binaries as well, e.g. config-reloader and admission-webhook? If yes, would it make sense to implement similarly to what Simon did [here](https://github.com/prometheus-operator/prometheus-operator/pull/6599/files)?

We could avoid code duplication that way.

Reply:
If we're not adding automatic gomemlimit to all binaries, then ignore this comment 😅

Reply:
Thank for your review. Done it.